sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/implementation/ReactorSenderTest.java

```diff
@@ -168,15 +207,22 @@ public void testLinkSizeSmallerThanMessageSize() {
             messageSerializer, Duration.ofSeconds(1), new ExponentialAmqpRetryPolicy(new AmqpRetryOptions()));
         ReactorSender spyReactorSender = spy(reactorSender);
 
-        doReturn(Mono.empty()).when(spyReactorSender).send(any(byte[].class), anyInt(), anyInt());
+        doReturn(Mono.empty()).when(spyReactorSender).send(any(byte[].class), anyInt(), anyInt(), isNull());
+
         StepVerifier.create(spyReactorSender.send(message))
             .verifyErrorSatisfies(throwable -> {
                 Assertions.assertTrue(throwable instanceof AmqpException);
                 Assertions.assertTrue(throwable.getMessage().startsWith("Error sending. Size of the payload exceeded "
                     + "maximum message size"));
             });
         verify(sender, times(1)).getRemoteMaxMessageSize();
-        verify(spyReactorSender, times(0)).send(any(byte[].class), anyInt(), anyInt());
+        verify(spyReactorSender, times(0)).send(any(byte[].class), anyInt(), anyInt(), isNull());
+    }
+
+    private DeliveryState getAmqpTransactionState(String transactionId) {
```

Comment:
This could be mocked rather than using a concrete implementation. Also, it is only used once.